addons/mass_mailing/models/mailing.py

```diff
@@ -362,8 +419,11 @@ def action_test(self):
         }
 
     def action_launch(self):
-        self.write({'schedule_type': 'now'})
-        return self.action_put_in_queue()
+        if self.testing_mailing_id:
+            self.testing_mailing_id.action_send_mailings(self)
+        else:
+            self.write({'schedule_type': 'now'})
+            self.action_put_in_queue()
```

Comment:
This feels a bit strange.
I think the safest thing would be to adapt the "_get_remaining_recipients" to compute a proper domain based on the fact that this mailing is part of a a/b testing.

This way is will directly work, for both "launch" and "schedule", but also when calling "action_put_in_queue" directly from the code.